lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -389,13 +392,17 @@ func AddInvoice(ctx context.Context, cfg *AddInvoiceConfig,
 		}
 
 		totalHopHints := len(invoice.RouteHints)
-		if invoice.Private {
+		if invoice.Private || invoice.HintsTarget > 0 {
 			totalHopHints = maxHopHints
 		}
 
 		hopHintsCfg := newSelectHopHintsCfg(cfg, totalHopHints)
+		nHintsTarget := 0
```

Comment:
Looks like we can add the numHopHints field to `SelectHopHintsCfg` too, and no need to pass it to `PopulateHopHints`.